File: 2021/day-15/main_test.py
import unittest

import main


class TestCase(unittest.TestCase):
    def setUp(self):
        self.map = main.read_input('input_test.txt')

    def test_part_one(self):
        self.assertEqual(main.part_one(self.map), 40)

    def test_part_two(self):
        self.assertEqual(main.part_two(self.map), 315)

File: 2022/day-04/main_test.go
package main

import "testing"

func TestPartOne(t *testing.T) {
	input, err := readLines("input_test.txt")
	if err != nil {
		panic(err)
	}

	got := partOne(input)
	want := 2
	if got != want {
		t.Errorf("got %v, wanted %v", got, want)
	}
}

func TestPartTwo(t *testing.T) {
	input, err := readLines("input_test.txt")
	if err != nil {
		panic(err)
	}

	got := partTwo(input)
	want := 4
	if got != want {
		t.Errorf("got %v, wanted %v", got, want)
	}
}
